Simplenote/src/main/java/com/automattic/simplenote/NotesActivity.java

```diff
@@ -771,11 +797,15 @@ public void onNoteSelected(String noteID, int position, String matchOffsets, boo
             arguments.putString(NoteEditorFragment.ARG_ITEM_ID, noteID);
             arguments.putBoolean(NoteEditorFragment.ARG_MARKDOWN_ENABLED, isMarkdownEnabled);
 
-            if (matchOffsets != null)
+            if (matchOffsets != null) {
                 arguments.putString(NoteEditorFragment.ARG_MATCH_OFFSETS, matchOffsets);
+            }
 
             Intent editNoteIntent = new Intent(this, NoteEditorActivity.class);
             editNoteIntent.putExtras(arguments);
+            if (mNoteListFragment.isHidden()) {
+                editNoteIntent.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
+            }
```

Comment:
Can you elaborate on the animation this code is removing?  You mention in the commit message that this code makes the animation smoother when transitioning from full-screen landscape to portrait.  When testing this code with a tablet with full-screen mode enabled, I'm not seeing any difference when rotating from landscape to portrait with or without this code.  I've tested on an emulator and device with the same results.

Reply:
According to [Android Developer Docs}(https://developer.android.com/reference/android/content/Intent#FLAG_ACTIVITY_NO_ANIMATION):
> If set in an Intent passed to Context#startActivity, this flag will prevent the system from applying an activity transition animation to go to the next activity state. This doesn't mean an animation will never run -- if another activity change happens that doesn't specify this flag before the activity started here is displayed, then that transition will be used. This flag can be put to good use when you are going to do a series of activity operations but the animation seen by the user shouldn't be driven by the first activity change but rather a later one.

When the user rotates the device back from full screen landscape editing to portrait, the `EditNoteActivity` will be opened. I wanted to prevent the device from playing 1. the rotation animation, then 2. the activity transition animation then 3. show the `NoteEditorActivity`. I wanted to make sure the transition was smooth and the device only played the rotation animation then showed the `NoteEditorActivity` as if no activity transition happened at all.

To be honest however, I hardly noticed any difference as well. Perhaps because it happens so fast, the devices we are using are quick, or the rotation animation overrides the activity transition animation on certain devices.

I am not entirely sure, but I thought it would be safer to explicitly specify the `FLAG_ACTIVITY_NO_ANIMATION` flag in the intent, making sure that the activity transition animation in this particular case does not happen. I thought this would be safer than assuming it will not make any difference on all devices or it will not be noticeable on slow devices.

That was my reasoning there.

Reply:
Thanks for the details about your reasoning there.  I don't see any difference either way so I guess we can keep it.